File: leetcode/2481.minimum-cuts-to-divide-a-circle/solution.cpp
#include "../../common.hpp"
// https://leetcode.com/problems/minimum-cuts-to-divide-a-circle/

class Solution {
public:
  static int numberOfCuts(int n) { return n == 1 ? 0 : (n & 1 ? n : (n >> 1)); }
};

int main(int argc, char *argv[]) {
  Solution s;
  DEBUG(s.numberOfCuts(5));
  return 0;
}

File: leetcode/2481.minimum-cuts-to-divide-a-circle/solution.rs
// https://leetcode.com/problems/minimum-cuts-to-divide-a-circle/
impl Solution {
    pub fn number_of_cuts(n: i32) -> i32 {
        if n == 1 {0} else if n % 2 == 0 {n/2} else {n}
    }
}

